#6289: Add new example to XFAIL documentation (#6685)
pytest-dev/pytest: doc/en/skipping.rst

    def test_function():
        ...

> This test will run but no traceback will be reported when it fails. Instead terminal

Reviewer: ```suggestion
This test will run but no traceback will be reported when it fails. Instead, terminal
```

reporting will list it in the "expected to fail" (``XFAIL``) or "unexpectedly
passing" (``XPASS``) sections.

> Alternatively, you can mark tests as ``XFAIL`` from within a test or setup function

Reviewer: I think the previous phrasing was slightly better, because the new one might imply that you can skip test A from within test B.

I would phrase it like this though:

``
 Alternatively, you can also mark a test as ``XFAIL`` from within the test or its setup function
imperatively:
```

Author: Updated PR based on feedback

        if not valid_config():
            pytest.xfail("failing configuration (but should work)")

.. code-block:: python

    def test_function2():
        import slow_module

        if slow_module.slow_function():
            pytest.xfail("slow_module taking too long")

These two examples illustrate situations where you don't want to check for a condition
at the module level, which is when a condition would otherwise be evaluated for marks.

This will make ``test_function`` ``XFAIL``. Note that no other code is executed after

> ``pytest.xfail`` call, differently from the marker. That's because it is implemented

Reviewer: ```suggestion
the ``pytest.xfail`` call, differently from the marker. That's because it is implemented
```

        if not valid_config():
            pytest.xfail("failing configuration (but should work)")

.. code-block:: python

    def test_function2():
        import slow_module

        if slow_module.slow_function():
            pytest.xfail("slow_module taking too long")

These two examples illustrate situations where you don't want to check for a condition
at the module level, which is when a condition would otherwise be evaluated for marks.

> They will unconditionally make the two examples ``XFAIL``. Note that no other code is executed

Reviewer: I know this is from before, but I'm confused by the "unconditionally" here -- the `pytets.xfail` is literally under an `if`. Maybe I'm misunderstanding?

Author: I was confused by it as well, but didn't want to remove without proper justification. I'll update with the "unconditionally" removed, and leave it to you for approval.

Author: Updated PR based on feedback

> after a ``pytest.xfail`` call, which is different from a marker. This is because current

Reviewer: I'd leave the last sentence as-is. I don't think there are any plans to change this? Nor can I think of any other possible implementation. And also it would be a breaking change to change anyway.

Author: Updated PR based on feedback

> Both ``XFAIL`` and ``XPASS`` don't fail the test suite unless the ``strict`` keyword-only

Reviewer: ```suggestion
Both ``XFAIL`` and ``XPASS`` don't fail the test suite by default. You can change this by setting the ``strict`` keyword-only parameter to ``True``:
```

Author: Updated PR based on feedback
